### PR TITLE
Remove xpack.searchable.snapshot.cache.size setting

### PR DESCRIPTION
**Pull request by tlrx**
The `xpack.searchable.snapshot.cache.size` setting was introduced to limit the total size of the disk based cache on Cold tier in the hope that it could be reused in a for or other for the Frozen tier. Then we agreed that this cache should be unbounded for the Cold tier and finally we implemented the Frozen tier with a different cache implementation (yet the Frozen tier uses this unbounded cache for caching some parts of metadata files). Limiting the size of this cache is now useless for production code and can be removed. It allows to remove some burden in the tests that used to randomize this setting.

Note that this setting was never documented.

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/snapshots.asciidoc [hunk 1]
```diff
@@ -143,3 +143,16 @@ deprecated starting 7.10.0 and will be removed in 8.0.0.
 *Impact* +
 Use the <<repositories-metering-apis,Repositories Metering APIs>>.
 ====
+
+.The `xpack.searchable.snapshot.cache.size` node-level setting has been removed
```
Comment (ywelsch) [inline docs/reference/migration/migrate_8_0/snapshots.asciidoc:147]: I don't think we should document removal of things that were never documented in the first place and only there for internal use.

**Review by ywelsch — COMMENTED** [review]
I've left one comment, o.w. looking good.

File: docs/reference/migration/migrate_8_0/snapshots.asciidoc (hunk 1 above)
Comment (tlrx) [reply docs/reference/migration/migrate_8_0/snapshots.asciidoc]: Yeah, I wasn't sure about this. I reverted the doc change.

**Review by ywelsch — APPROVED** [review]
LGTM. Thank you!

**Comment by tlrx** [comment]
Thanks Yannick

**Pull request merged**